engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -117,26 +112,24 @@ public void onEnter(VirtualFrame frame) {
         if (!isTopFrame(entryCallTarget)) {
           return;
         }
-        onEnterImpl();
-      }
 
-      @CompilerDirectives.TruffleBoundary
-      private void onEnterImpl() {
-        UUID nodeId = getNodeId(context.getInstrumentedNode());
-        var result = callbacks.findCachedResult(nodeId);
+        Node node = context.getInstrumentedNode();
+        UUID nodeId = getNodeId(node);
+
+        callbacks.onEnter(nodeId);
+        Object result = executionSupport.findCachedResult(frame.materialize(), node, nodeId);
```

Comment:
No, we want a `Node` here. A `Node` that will compute the desired expression and assign its result to the `result` variable. The `Callbacks` interface doesn't have to change at all.